frontend/client/src/components/PendingOperationButton.js

```diff
@@ -18,20 +18,39 @@ const PendingOperationButton = (props) => {
   const operation = props.operation || (() => {})
   const disabled = props.disabled || false
 
-  const startOperation = async () => {
-    setIsOperationPending(true)
+  // Flag to keep track of whether component is mounted. We check this flag in runOperation before setting the button's state,
+  // in order to prevent cases where we mistakenly try to update the state when the button is no longer mounted, which was
+  // happening relatively frequently in i.e. automatic-modal-closes and triggering the following warning:
+  // "Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application.
+  // To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function."
+  var componentIsMounted = false
```

Comment:
Can this be part of the state? I.e. change to `const [componentIsMounted, setComponentIsMounted] = useState(false)`

Reply:
Why should it be?